<a href="https://colab.research.google.com/github/SediKOR/Ybigta_assignment/blob/main/%ED%9A%8C%EA%B7%80%EB%B6%84%EC%84%9D_%EC%9D%B4%EC%84%B8%EC%A7%84.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 과제

- 'Student_Performance.csv'를 이용해 다중선형회귀분석을 진행하세요!


In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [4]:
# 데이터 불러오기
data = pd.read_csv('drive/MyDrive/Student_Performance.csv', index_col = 0)
data.head()

Study_Hours  Prev_Scores  Sleep_Hours  Sample_Practice    PI  \
0            7           99            9                1  91.0   
1            4           82            4                2  65.0   
2            8           51            7                2  45.0   
3            5           52            5                2  36.0   
4            7           75            8                5  66.0   

   Extracurriculars  
0                 1  
1                 0  
2                 1  
3                 1  
4                 0

2.1. 반응변수(Y)를 PI로 하고, 설명변수(X)들을 그 외 모든 변수로 설정한 다중선형회귀분석을 진행하고 summary를 print하세요.

In [18]:
import statsmodels.api as sm
from statsmodels.formula.api import ols

#1 - formula 명시하지 않는 방법
Y = data.iloc[:, -2]
X = data.iloc[:, data.columns != 'PI']
X = sm.add_constant(X)

model = sm.OLS(Y, X)
res = model.fit()

print(res.summary())


                            OLS Regression Results                            
Dep. Variable:                     PI   R-squared:                       0.989
Model:                            OLS   Adj. R-squared:                  0.989
Method:                 Least Squares   F-statistic:                 1.757e+05
Date:                Sat, 15 Jul 2023   Prob (F-statistic):               0.00
Time:                        06:19:11   Log-Likelihood:                -21307.
No. Observations:               10000   AIC:                         4.263e+04
Df Residuals:                    9994   BIC:                         4.267e+04
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
const              -34.0756      0.127  

2.2. 반응변수를 PI로 하고, 설명변수들을 Study_Hours로 설정한 단순선형회귀분석을 진행하고 summary를 print하세요.

In [17]:
import statsmodels.api as sm
from statsmodels.formula.api import ols

#1 - formula 명시하지 않는 방법
Y_2 = data.iloc[:, -2]
X_2 = data.iloc[:, data.columns == 'Study_Hours']
X_2 = sm.add_constant(X)

model_2 = sm.OLS(Y_, X_)
res_2 = model_2.fit()

print(res_.summary())


                            OLS Regression Results                            
Dep. Variable:                     PI   R-squared:                       0.989
Model:                            OLS   Adj. R-squared:                  0.989
Method:                 Least Squares   F-statistic:                 1.757e+05
Date:                Sat, 15 Jul 2023   Prob (F-statistic):               0.00
Time:                        06:19:08   Log-Likelihood:                -21307.
No. Observations:               10000   AIC:                         4.263e+04
Df Residuals:                    9994   BIC:                         4.267e+04
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
const              -34.0756      0.127  

2.3. (2.1)과 (2.2)의 분석을 비교하고 Study_Hours와 PI의 관계를 짧게, 자유롭게 설명해보세요.

설명: 첫번째 회귀분석 결과의 R-squared는 0.989로 매우 높은 값을 가지고 있다. 이는 독립변수가 종속변수를 잘 설명했다고 할 수 있다. coef를 보면, Study_Hours가 증가할수록 PI도 증가한다는 것을 알 수 있으며 t통계량을 통해 유의미함이 증명되었다. 두번째 회귀분석 결과의 R-squared는 0.140으로 낮은 값을 가지고 있다. 이는 Study_Hours만 가지고 PI의 변동을 설명하기에 한계가 존재한다는 뜻이다. coef는 2.7731로, Study_Hours가 증가할수록 PI도 증가하며 t통계량을 통해 유의미함이 증명되었으나 Study_Hours만 가지고는 판단할 수 없다. 따라서 Study_Hours와 다른 설명변수들을 고려해야 PI를 판단할 수 있을 것이다.